foundry/foundry.py

```diff
@@ -100,19 +114,35 @@ def load(self, name, download=True, globus=True, verbose=False, **kwargs):
         -------
             self
         """
+        res = []
         # MDF specific logic
-        res = self.forge_client.match_field(
-            "mdf.organizations", "foundry"
-        ).match_resource_types("dataset")
-        res = res.match_field("mdf.source_id", name).search()
+        # Handle DOI inputs
+        if name.startswith("10.") and provider == "MDF":
+            res = (
+                self.forge_client.match_dois(name)
+                .match_resource_types("dataset")
+                .match_field("mdf.organizations", "foundry")
+                .search()
+            )
 
-        # TODO: if object empty, handle
-        res = res[0]
-        res["dataset"] = res["projects"]["foundry"]
-        res["dataset"]["type"] = res["dataset"]["package_type"]
-        del res["projects"]["foundry"]
+        # Handle MDF source_ids
+        else:
+            print("source_id")
+            res = (
+                self.forge_client.match_field("mdf.organizations", "foundry")
+                .match_resource_types("dataset")
+                .match_field("mdf.source_id", name)
+                .search()
+            )
 
-        self = Foundry(**res)
+        if not res:
+            return self
+        else:
+            res = res[0]
+            res["dataset"] = res["projects"]["foundry"]
+            res["dataset"]["type"] = res["dataset"]["package_type"]
```

Comment:
for my own learning, could you explain what's going on here? why would we need to restructure the result like this?

Reply:
I had to think on this a bit too...which means I should be adding more comments :) 

Lines 141 onward map the Forge return value into the FoundryMetadata Pydantic dictionary. The def for the Pydantic is in models.py

Reply:
adding comment